dlt/common/runtime/telemetry.py

```diff
@@ -115,3 +119,22 @@ def _track(success: bool) -> None:
         return _wrap  # type: ignore
 
     return decorator
+
+
+def with_dataset_access_telemetry(
```

Comment:
this is good name for a decorator and this is a regular function. so the name should be an action ie. track_pipeline_create_dataset or smth

as mentioned above: this function fits track module better